pkgs/servers/web-apps/funkwhale/default.nix

```diff
@@ -0,0 +1,158 @@
+{ lib
+, callPackage
+, fetchFromGitLab
+, python3
+, mkYarnPackage
+, jq
+, ffmpeg
+}:
+
+let
+  pname = "funkwhale";
+  version = "1.2.0";
+
+  python = python3.override {
+    packageOverrides = self: super: {
+      django = super.django_3;
+    };
+  };
+
+  src = fetchFromGitLab {
+    domain = "dev.funkwhale.audio";
+    owner = pname;
+    repo = pname;
+    rev = version;
+    hash = "sha256:1y2zkcc5il4shhcxxwpyn9yjlw46b5zydg868s99yxwa1l83y2cz";
+  };
+
+  frontend = mkYarnPackage {
+    pname = "funkwhale-frontend";
+    inherit version;
+    src =  src + "/front";
+    packageJSON = src + "/front/package.json";
+    yarnLock = src + "/front/yarn.lock";
+
+    # patch shebangs that patchshebangs can't handle
+    postPatch = ''
+      sed -i 's/env -S bash -eux/env bash\nset -eux/' scripts/*.sh
+      substituteInPlace scripts/utils.sh \
+        --replace "yarn bin" "yarn --offline bin"
+      substituteInPlace scripts/fix-fomantic-css.sh \
+        --replace "find" "find -L"
+      patchShebangs scripts
+    '';
+
+    nativeBuildInputs = [
+      jq
+      python3
+    ];
+
+    buildPhase = ''
+      (
+        cd deps/front
+        rm node_modules
+        ln -s ../../node_modules node_modules
+      )
+      yarn --offline run fix-fomantic-css
+
+      # Error: No PostCSS Config found in: /build/front/node_modules/fomantic-ui-css/tweaked
+      echo "module.exports = {};" > node_modules/fomantic-ui-css/tweaked/postcss.config.js
+
+      yarn --offline build
+    '';
+
+    installPhase = ''
+      cp -rv deps/front/dist $out
+    '';
+
+    distPhase = "true";
+  };
+
+  path = lib.makeBinPath [
+    ffmpeg
+  ];
+in
+python.pkgs.buildPythonApplication rec {
+  inherit pname version src;
+  format = "other";
+
+  dontBuild = true;
+
+  propagatedBuildInputs = with python.pkgs; [
+    PyLD
+    aiohttp
+    arrow
+    asgiref
+    bleach
+    boto3
+    celery
+    channels
+    channels-redis
+    click
+    cryptography
+    django-allauth
+    django-auth-ldap
+    django-cacheops
+    django-cleanup
+    django-cors-headers
+    django-dynamic-preferences
+    django-environ
+    django-filter
+    django-oauth-toolkit
+    django-redis
+    django-rest-auth
+    django-storages
+    django-versatileimagefield
+    django
+    djangorestframework
+    feedparser
+    kombu
+    ldap
+    markdown
+    musicbrainzngs
+    mutagen
+    persisting-theory
+    pillow
+    psycopg2
+    pydub
+    pymemoize
+    pyopenssl
+    magic
+    pytz
+    redis
+    requests
+    requests-http-signature
+    service-identity
+    unicode-slugify
+    uvicorn
+    watchdog
+  ];
+
+  installPhase = ''
+    mkdir -p $out/lib
+    cp -r $src/api $out/lib/funkwhale
+
+    chmod +w $out/lib/funkwhale
+    cp -r ${frontend}/ $out/lib/funkwhale/front
```

Comment:
This breaks eval tests 🤷

Reply:
Looks like IFD to me.

```
error: cannot read '/build/store/qj51dlf6gibab5hm4a6h22sgicsq42zf-source/front/package.json', since path '/build/store/3khl9w0x8hx0v034p7ypsj82ygv66761-source.drv' is not valid

       at /nix/store/pyzjdbyr1shws1hhy28h94gs4drimlc3-source/lib/trivial.nix:282:24:

          281|   importJSON = path:
          282|     builtins.fromJSON (builtins.readFile path);
             |                        ^
          283|

       … while evaluating 'importJSON'
```